doc/nrf/releases_and_maturity/abi_compatibility.rst

```diff
@@ -0,0 +1,40 @@
+.. _abi_compatibility:
+
+ABI Compatibility
+*****************
+
+.. contents::
+   :local:
+   :depth: 2
+
+Application Binary Interface (ABI) compatibility defines how software components, like libraries and applications, interacts at the machine code level through their low-level binary interface, including:
+
+* Function-calling conventions
+* Data structure layouts in memory
+* Exception handling mechanisms
+* Register usage conventions
+
+When ABI compatibility is maintained, binaries of one component can interface correctly with another without needing recompilation.
+For example, adding a new function to a library is typically an ABI-compatible change, as existing binaries remain functional.
+However, changes that affect data structure layouts, such as altering field order or size, break ABI compatibility as they change the memory layout expected by existing binaries.
+
+ABI Compatibility Matrix for the for the nrf54h20_soc_binary bundle
```

Comment:
I would add a paragraph to this section explaining that ABI compatibility for nrf54h20_soc_binary means that the System Controller or Secure Domain binaries do not need to be recompiled when the Application or Radio binary gets recompiled, as long as the Application and Radio are based on compatible NCS versions described in this section.